packages/gestalt/src/boxTransforms.js

```diff
@@ -0,0 +1,648 @@
+
+type PaddingFunctor = Functor<Padding>;
+
+const paddingX: PaddingFunctor = bind(rangeWithoutZero('paddingX'), whitespace);
+const paddingY: PaddingFunctor = bind(rangeWithoutZero('paddingY'), whitespace);
+const padding: PaddingFunctor = union(paddingX, paddingY);
+
+const smPaddingX: PaddingFunctor = bind(
+  rangeWithoutZero('smPaddingX'),
+  whitespace
+);
+const smPaddingY: PaddingFunctor = bind(
+  rangeWithoutZero('smPaddingY'),
+  whitespace
+);
+const smPadding: PaddingFunctor = union(smPaddingX, smPaddingY);
+
+const mdPaddingX: PaddingFunctor = bind(
+  rangeWithoutZero('mdPaddingX'),
+  whitespace
+);
+const mdPaddingY: PaddingFunctor = bind(
+  rangeWithoutZero('mdPaddingY'),
+  whitespace
+);
+const mdPadding: PaddingFunctor = union(mdPaddingX, mdPaddingY);
+
+const lgPaddingX: PaddingFunctor = bind(
+  rangeWithoutZero('lgPaddingX'),
+  whitespace
+);
+const lgPaddingY: PaddingFunctor = bind(
+  rangeWithoutZero('lgPaddingY'),
+  whitespace
+);
+const lgPadding: PaddingFunctor = union(lgPaddingX, lgPaddingY);
+
+/* ***************************************** */
+
+const position: Functor<Position> = mapping({
+  absolute: layout.absolute,
+  relative: layout.relative,
+  fixed: layout.fixed,
+  // default: static
+});
+const right: Functor<boolean> = toggle(layout.right0);
+const rounding = getRoundingStyle;
+const top: Functor<boolean> = toggle(layout.top0);
+const userSelect: Functor<UserSelect> = mapping({
+  none: styles.userSelectNone,
+  // default: auto
+});
+const width: Functor<Dimension> = w => fromInlineStyle({ width: w });
+const wrap: Functor<boolean> = toggle(layout.flexWrap);
+const dangerouslySetInlineStyle: Functor<DangerouslySetInlineStyle> = value => {
+  // eslint-disable-next-line no-underscore-dangle
+  return value && value.__style ? fromInlineStyle(value.__style) : identity();
+};
+const zIndex: Functor<?Indexable> = value => {
+  if (!value) {
+    return identity();
+  }
+  return fromInlineStyle({ zIndex: value.index() });
+};
+
+/*
+
+These functions are legacy. I'd like to get rid of most of this file's dependency on importing `./style.js` directly once these are removed.
+
+*/
+
+const prefix = (pre: string) => mapClassName(name => `${pre}${name}`);
+const displayFunctor = value => {
+  switch (value) {
+    case 'flex':
+      return fromClassName('DisplayFlex', 'DirectionRow');
+    case 'flexColumn':
+      return fromClassName('DisplayFlex', 'DirectionColumn');
+    case 'inlineBlock':
+      return fromClassName('DisplayInlineBlock');
+    case 'visuallyHidden':
+      return fromClassName('DisplayVisuallyHidden');
+    case false:
+      return fromClassName('DisplayNone');
+    default:
+      /* block */
+      return fromClassName('DisplayBlock');
+  }
+};
+const columnRange = range('Col');
+
+/*
+
+It's preferable to put new properties into that object directly just so it's easier to read.
+
+*/
+
+export const propToFn = {
+  xs: (value: ?ResponsiveProps): Style => {
```

Comment:
contrary to the other fields, Flow gave me a lot of grief when I tried to abstract these to a `buildResponsiveStyles` helper, so they remain inline